### PR TITLE
TRUS-2475: Add Individual Protector

### PR DESCRIPTION
**Pull request by BriWak**

### CODE REVIEW

File: app/utils/print/IndividualProtectorPrintHelper.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  def apply(userAnswers: UserAnswers, provisional: Boolean, settlorName: String)(implicit messages: Messages) = {
+
+    val bound = answerRowConverter.bind(userAnswers, settlorName, countryOptions)
+
+    val add: Seq[AnswerRow] = Seq(
+        bound.nameQuestion(NamePage, "individualProtector.name", controllers.individual.routes.NameController.onPageLoad(NormalMode).url),
+        bound.yesNoQuestion(DateOfBirthYesNoPage, "individualProtector.dateOfBirthYesNo", controllers.individual.routes.DateOfBirthYesNoController.onPageLoad(NormalMode).url),
+        bound.dateQuestion(DateOfBirthPage, "individualProtector.dateOfBirth", controllers.individual.routes.DateOfBirthController.onPageLoad(NormalMode).url),
+        bound.yesNoQuestion(NationalInsuranceNumberYesNoPage, "individualProtector.nationalInsuranceNumberYesNo", controllers.individual.routes.NationalInsuranceNumberYesNoController.onPageLoad(NormalMode).url),
+        bound.ninoQuestion(NationalInsuranceNumberPage, "individualProtector.nationalInsuranceNumber", controllers.individual.routes.NationalInsuranceNumberYesNoController.onPageLoad(NormalMode).url),
```
Comment (jackheslop96) [inline app/utils/print/IndividualProtectorPrintHelper.scala:39]: Should be redirecting to NationalInsuranceNumberController

File: app/utils/print/IndividualProtectorPrintHelper.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        bound.dateQuestion(StartDatePage, "individualProtector.startDate", controllers.individual.routes.StartDateController.onPageLoad().url)
+      ).flatten
+
+    val amend: Seq[AnswerRow] = Seq(
+      bound.nameQuestion(NamePage, "individualProtector.name", controllers.individual.routes.NameController.onPageLoad(CheckMode).url),
+      bound.yesNoQuestion(DateOfBirthYesNoPage, "individualProtector.dateOfBirthYesNo", controllers.individual.routes.DateOfBirthYesNoController.onPageLoad(CheckMode).url),
+      bound.dateQuestion(DateOfBirthPage, "individualProtector.dateOfBirth", controllers.individual.routes.DateOfBirthController.onPageLoad(CheckMode).url),
+      bound.yesNoQuestion(NationalInsuranceNumberYesNoPage, "individualProtector.nationalInsuranceNumberYesNo", controllers.individual.routes.NationalInsuranceNumberYesNoController.onPageLoad(CheckMode).url),
+      bound.ninoQuestion(NationalInsuranceNumberPage, "individualProtector.nationalInsuranceNumber", controllers.individual.routes.NationalInsuranceNumberYesNoController.onPageLoad(CheckMode).url),
```
Comment (jackheslop96) [inline app/utils/print/IndividualProtectorPrintHelper.scala:56]: Should be redirecting to NationalInsuranceNumberController

File: conf/messages.en [hunk 3]
```diff
@@ -45,6 +45,24 @@ site.textarea.char_limit = (Limit is {0} characters)
...
+site.address.uk.postcode = Postcode
+site.address.uk.postcode.hint = For example AB1 1AB
+
+address.country.select = Select a country
+site.address.nonUk.title = What is the address?
+site.address.nonUk.heading = What is {0}’s address?
+site.address.nonUk.line1 = Address line 1
+site.address.nonUk.line2 = Address line 2
+site.address.nonUk.line3 = Address line 3 (optional)
+site.address.nonUk.country = Country
```
Comment (jackheslop96) [inline conf/messages.en:65]: Duplicated

File: conf/messages.en [hunk 4]
```diff
@@ -118,6 +136,138 @@ entities.protector = Protectors
...
+individualProtector.dateOfBirth.error.required.two = Date of birth must include a {0} and a {1}
+individualProtector.dateOfBirth.error.required = Date of birth must include a {0}
+individualProtector.dateOfBirth.error.invalid = Enter a real date of birth
+individualProtector.dateOfBirth.error.future = Date of birth must be today or in the past
+individualProtector.dateOfBirth.error.past = Date of birth must be after 31 December 1499
+
+individualProtector.nationalInsuranceNumberYesNo.checkYourAnswersLabel = Do you know {0}’s National Insurance number?
+individualProtector.nationalInsuranceNumberYesNo.title = Do you know the individual’s National Insurance number?
+individualProtector.nationalInsuranceNumberYesNo.heading = Do you know {0}’s National Insurance number?
+individualProtector.nationalInsuranceNumberYesNo.error.required = Select yes if you know the protector’s National Insurance number
```
Comment (jackheslop96) [inline conf/messages.en:174]: These don't feel consistent to me. Is it the individual or the protector?

File: app/utils/print/IndividualProtectorPrintHelper.scala (hunk 1 above)
Comment (BriWak) [reply app/utils/print/IndividualProtectorPrintHelper.scala]: Fixed now

File: app/utils/print/IndividualProtectorPrintHelper.scala (hunk 2 above)
Comment (BriWak) [reply app/utils/print/IndividualProtectorPrintHelper.scala]: Fixed now

File: conf/messages.en [hunk 5]
```diff
@@ -118,6 +136,138 @@ entities.protector = Protectors
...
+individualProtector.passportDetails.number.hint = For example, 502135326
+individualProtector.passportDetails.number.error.required = Enter the protector’s passport number
+individualProtector.passportDetails.number.error.length = The protector’s passport number must be 30 characters or less
+individualProtector.passportDetails.number.error.invalid = Passport number can only include letters and numbers
+individualProtector.passportDetails.expiryDate = Expiry date
+individualProtector.passportDetails.expiryDate.hint = Expiry date can be in the past. For example, 31 3 1980
+individualProtector.passportDetails.expiryDate.error.required.all = Enter the passport expiry date
+individualProtector.passportDetails.expiryDate.error.required.two = Passport expiry date must include a {0} and {1}
+individualProtector.passportDetails.expiryDate.error.required = Passport expiry date must include {0}
+individualProtector.passportDetails.expiryDate.error.invalid = Enter a real date when the individual’s passport expires
```
Comment (jackheslop96) [inline conf/messages.en:220]: individual, protector or individual protector? Unfortunately I can't seem to see protectors on the task list in the prototype so I can't check

File: conf/messages.en (hunk 3 above)
Comment (BriWak) [reply conf/messages.en]: Fixed now

File: conf/messages.en [hunk 6]
```diff
@@ -118,6 +136,138 @@ entities.protector = Protectors
...
+individualProtector.idCardDetails.number.hint = For example, 502135326
+individualProtector.idCardDetails.number.error.required = Enter the protector’s ID card number
+individualProtector.idCardDetails.number.error.length = The protector’s ID card number must be 30 characters or less
+individualProtector.idCardDetails.number.error.invalid = ID card number can only include letters and numbers
+individualProtector.idCardDetails.expiryDate = Expiry date
+individualProtector.idCardDetails.expiryDate.hint = Expiry date can be in the past. For example, 31 3 1980
+individualProtector.idCardDetails.expiryDate.error.required.all = Enter the ID card expiry date
+individualProtector.idCardDetails.expiryDate.error.required.two = ID card expiry date must include a {0} and {1}
+individualProtector.idCardDetails.expiryDate.error.required = ID card expiry date must include a {0}
+individualProtector.idCardDetails.expiryDate.error.invalid = Enter a real date when the individual’s ID card expires
```
Comment (jackheslop96) [inline conf/messages.en:245]: Again, using 'individual' when 'protector' has been used elsewhere. May be worth double checking with Rachel

File: app/utils/print/IndividualProtectorPrintHelper.scala [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import pages.individual._
+import play.api.i18n.Messages
+import utils.countryOptions.CountryOptions
+import viewmodels.{AnswerRow, AnswerSection}
+
+class IndividualProtectorPrintHelper @Inject()(answerRowConverter: AnswerRowConverter,
+                                               countryOptions: CountryOptions
+                                            ) {
+
+  def apply(userAnswers: UserAnswers, provisional: Boolean, settlorName: String)(implicit messages: Messages) = {
```
Comment (jackheslop96) [inline app/utils/print/IndividualProtectorPrintHelper.scala:30]: protectorName

File: conf/messages.en (hunk 6 above)
Comment (BriWak) [reply conf/messages.en:245]: These are the same format settlor uses so I'm assuming they are correct?

File: test/controllers/individual/AddressYesNoControllerSpec.scala [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+import scala.concurrent.Future
+
+class AddressYesNoControllerSpec extends SpecBase with MockitoSugar {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new YesNoFormProvider()
+  val form = formProvider.withPrefix("individualProtector.addressYesNo")
+  val trusteeName = "FirstName LastName"
```
Comment (jackheslop96) [inline test/controllers/individual/AddressYesNoControllerSpec.scala:45]: protectorName

File: test/controllers/individual/DateOfBirthControllerSpec.scala [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+class DateOfBirthControllerSpec extends SpecBase with MockitoSugar {
+
+  val formProvider = new DateOfBirthFormProvider()
+  private def form = formProvider.withPrefix("individualProtector.dateOfBirth")
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val validAnswer = LocalDate.now(ZoneOffset.UTC)
+  val trusteeName = "FirstName LastName"
```
Comment (jackheslop96) [inline test/controllers/individual/DateOfBirthControllerSpec.scala:47]: protectorName

File: test/controllers/individual/LiveInTheUkYesNoControllerSpec.scala [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+import scala.concurrent.Future
+
+class LiveInTheUkYesNoControllerSpec extends SpecBase with MockitoSugar {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new YesNoFormProvider()
+  val form = formProvider.withPrefix("individualProtector.liveInTheUkYesNo")
+  val trusteeName = "FirstName LastName"
```
Comment (jackheslop96) [inline test/controllers/individual/LiveInTheUkYesNoControllerSpec.scala:45]: protectorName

File: test/controllers/individual/NameControllerSpec.scala [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+import scala.concurrent.Future
+
+class NameControllerSpec extends SpecBase with MockitoSugar {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new NameFormProvider()
+  val form = formProvider.withPrefix("individualProtector.name")
+  val trusteeName = Name("FirstName", None, "LastName")
```
Comment (jackheslop96) [inline test/controllers/individual/NameControllerSpec.scala:43]: protectorName

File: test/controllers/individual/NationalInsuranceNumberControllerSpec.scala [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+import scala.concurrent.Future
+
+class NationalInsuranceNumberControllerSpec extends SpecBase with MockitoSugar {
+
+  def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new NationalInsuranceNumberFormProvider()
+  val form = formProvider.withPrefix("individualProtector.nationalInsuranceNumber")
+
+  val trusteeName = Name("FirstName", None, "LastName")
```
Comment (jackheslop96) [inline test/controllers/individual/NationalInsuranceNumberControllerSpec.scala:44]: protectorName

File: test/controllers/individual/NationalInsuranceNumberYesNoControllerSpec.scala [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+import repositories.PlaybackRepository
+import views.html.individual.NationalInsuranceNumberYesNoView
+
+import scala.concurrent.Future
+
+class NationalInsuranceNumberYesNoControllerSpec extends SpecBase with MockitoSugar {
+
+  val formProvider = new YesNoFormProvider()
+  val form = formProvider.withPrefix("individualProtector.nationalInsuranceNumberYesNo")
+  val trusteeName = "FirstName LastName"
```
Comment (jackheslop96) [inline test/controllers/individual/NationalInsuranceNumberYesNoControllerSpec.scala:42]: protectorName

File: app/utils/print/IndividualProtectorPrintHelper.scala (hunk 7 above)
Comment (BriWak) [reply app/utils/print/IndividualProtectorPrintHelper.scala:30]: done

File: test/controllers/individual/AddressYesNoControllerSpec.scala (hunk 8 above)
Comment (BriWak) [reply test/controllers/individual/AddressYesNoControllerSpec.scala]: done

File: test/controllers/individual/DateOfBirthControllerSpec.scala (hunk 9 above)
Comment (BriWak) [reply test/controllers/individual/DateOfBirthControllerSpec.scala]: done

File: test/controllers/individual/LiveInTheUkYesNoControllerSpec.scala (hunk 10 above)
Comment (BriWak) [reply test/controllers/individual/LiveInTheUkYesNoControllerSpec.scala]: done

File: test/controllers/individual/NameControllerSpec.scala (hunk 11 above)
Comment (BriWak) [reply test/controllers/individual/NameControllerSpec.scala]: done

File: test/controllers/individual/NationalInsuranceNumberControllerSpec.scala (hunk 12 above)
Comment (BriWak) [reply test/controllers/individual/NationalInsuranceNumberControllerSpec.scala]: done

File: test/controllers/individual/NationalInsuranceNumberYesNoControllerSpec.scala (hunk 13 above)
Comment (BriWak) [reply test/controllers/individual/NationalInsuranceNumberYesNoControllerSpec.scala]: done

File: conf/messages.en (hunk 6 above)
Comment (jackheslop96) [reply conf/messages.en:245]: Could you double check with Rachel please? Doesn't seem right to refer to 'the protector' in all other error messages and then 'the individual' in this one

**Pull request merged**